Replace mocha+chai+sinon by Jest

Kureev opened the pull request:
For a painless merging `rnpm` in `react-native`'s core, we need to migrate our test stack to Jest.

**What's done:**
- [x] Replaced mocha + chai + sinon by Jest
- [x] Code coverage is now a Jest responsibility (it includes Istanbul by default)
- [x] Code coverage has been improved
- [x] Fixed some buggy edge cases

Kureev commented:
Rebased branch after merging in #95

grabbou commented on test/ios/getProjectConfig.spec.js:
```diff
@@ -1,31 +1,25 @@
-  before(() => {
-    mockFs({ testDir: projects });
-  });
+  beforeEach(() => mockFs({ testDir: projects }));
```
Do we need beforeEach in this case?

Kureev replied:
to be honest I haven't revisited test's logic, just replaced some technical aspects. But I can take a look

grabbou replied:
You changed before to be beforeEach so that's why I am asking ;)

Kureev replied:
You were right - we don't need these hooks anymore.

Kureev replied:
That was false positive. We need these hooks. As I already mentioned, that was the case because 
```js
typeof null === object
```
Now this behavior is fixed.

grabbou commented on test/android/findAndroidAppFolder.spec.js:
```diff
@@ -1,12 +1,11 @@
 describe('android::findAndroidAppFolder', () => {
-
-  before(() => mockFs({
+  beforeEach(() => mockFs({
```
Same here and in other suites, I think that could potentially slow down the test suite (all our functions are just performing reads so there's no need to re-mock every time)

Kureev commented:
before/afterEach hooks has been removed

Kureev commented:
As we discussed on Discord, I've added additional checks for `null` in all cases where it can be an issue. Also, reverted removing `beforeEach/afterEach` (false positive because of `null`).

Kureev commented:
100% test coverage, baby :tada:

grabbou commented:
LOL +175 −175 

LGTM
